actgen/models/inverse.py

```diff
@@ -0,0 +1,74 @@
+import torch
+
+from .. import nnutils
+
+class InverseModel(nnutils.Network):
+    """
+    Description:
+        Predicts the action distribution given a (state, next_state) pair.
+
+    Parameters:
+        - n_actions : Int
+            The number of actions or action dimensions
+        - params : dict
+            Dictionary of hyperparameters
+    """
+    def __init__(self, n_actions, params, discrete=True):
+        super(InverseModel, self).__init__()
+        self.discrete = discrete
+        self.body = torch.nn.Sequential(
+            torch.nn.Linear(params['latent_dim'] * 2, params['inv_layer_size']),
+            torch.nn.ReLU(),
+            torch.nn.Linear(params['inv_layer_size'], params['inv_layer_size']),
+            torch.nn.ReLU(),
+        )
+        if self.discrete:
+            self.log_pr_linear = torch.nn.Linear(params['inv_layer_size'], n_actions)
+            self.cross_entropy = torch.nn.CrossEntropyLoss()
+        else:
+            self.mean_linear = torch.nn.Linear(params['inv_layer_size'], n_actions)
+            self.log_std_linear = torch.nn.Linear(params['inv_layer_size'], n_actions)
+            self.log_sig_min = params['inv_log_std_min']
+            self.log_sig_max = params['inv_log_std_max']
+
+    def forward(self, z0, z1):
+        """
+        Compute the action distribution for two subsequent states.
+
+        If the action space is discrete, the output will be log probabilities,
```

Comment:
@camall3n the comments here say the output will be log probabilities, but it looks like `self.log_pr_linear` is just a linear layer, where is the `log` coming from?

Reply:
This is what people typically refer to as "logits", i.e. the input to a softmax layer. Since softmax applies an exponential to produce probabilities, the inputs are effectively log-probabilities.

Reply:
Note that there is no softmax layer here, so if you want normalized probabilities, you need to send the `log_pr_linear` through a softmax